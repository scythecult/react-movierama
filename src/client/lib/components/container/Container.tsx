import type { PropsWithChildren } from 'react';
import styles from './styles.module.css';

export const Container = (props: PropsWithChildren) => {
  const { children } = props;

  return <div className={styles.container}>{children}</div>;
};
