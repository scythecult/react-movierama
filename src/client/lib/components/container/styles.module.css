.container {
  perspective: 1000px;
  margin-bottom: 30px;
}
