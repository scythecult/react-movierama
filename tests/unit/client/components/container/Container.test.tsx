import { render } from '@testing-library/react';
import { Container } from '@/client/lib/components/container/Container';

describe('Container', () => {
  test('should correspond default layout', () => {
    const result = render(<Container />);

    expect(result.container).toMatchSnapshot();
  });

  test('should support children', () => {
    const result = render(<Container>TEST</Container>);

    expect(result.container).toMatchSnapshot();
  });
});
