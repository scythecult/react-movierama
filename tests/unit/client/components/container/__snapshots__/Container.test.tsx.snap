// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > should correspond default layout 1`] = `
<div>
  <div
    class="_container_72892e"
  />
</div>
`;

exports[`Container > should support children 1`] = `
<div>
  <div
    class="_container_72892e"
  >
    TEST
  </div>
</div>
`;
